velox/functions/lib/string/StringImpl.h

```diff
@@ -268,6 +268,15 @@ FOLLY_ALWAYS_INLINE bool md5_radix(
   return true;
 }
 
+/// Compute the SHA1 Hash.
+template <typename TOutString, typename TInString>
```

Comment:
Do we need these template parameters? Aren't they always arg_type<Varbinary> and out_type<Varbinary>?

Reply:
I don't know actually, I simply followed sha256 and sha512 methods: 
```c++
template <typename TOutString, typename TInString>
FOLLY_ALWAYS_INLINE void sha256(TOutString& output, const TInString& input) { ... }

template <typename TOutString, typename TInString>
FOLLY_ALWAYS_INLINE void sha512(TOutString& output, const TInString& input) { ... }
```